src/compat/db_vdb.c

```diff
@@ -2733,7 +2733,16 @@ do_recompile_and_execute_prepared_statement (DB_SESSION * session, PT_NODE * sta
       return er_errid ();
     }
 
-  assert (new_session->parser->set_host_var == 1);
+  if (new_session->parser->set_host_var == 0)
+    {
+      /* Cast host variable to expected domain, if not already casted in db_compile_statement. */
+      err = do_cast_host_variables_to_expected_domain (new_session);
+      if (err != NO_ERROR)
+	{
+	  return err;
+	}
+      new_session->parser->set_host_var = 1;
```

Comment:
I propose to move :
`new_session->parser->set_host_var = 1;`
inside of function:
do_cast_host_variables_to_expected_domain

It will prevent us forgetting setting the flag in other usages of the function.
Same is done with function 'pt_set_host_variables'